--- a/example/tkpasswd
+++ b/example/tkpasswd
@@ -4,7 +4,10 @@
 exec tclsh "$0" ${1+"$@"}
 
 package require Expect
-package require Tk
+if {[catch {package require Tk}]} {
+    puts stderr "This script require Tk to run. You can install the tk package for that."
+    exit 1
+}
 
 # tkpasswd - Change passwords using Expectk
 # Author: Don Libes, NIST, October 1, 1993
--- a/example/multixterm
+++ b/example/multixterm
@@ -312,7 +312,10 @@
 set versionDate "2004/06/29"
 
 package require Tcl
-catch {package require Tk} ;# early versions of Tk had no package
+if {[catch {package require Tk}]} {
+    puts stderr "This script require Tk to run. You can install the tk package for that."
+    exit 1
+}
 package require Expect
 
 proc exit1 {msg} {
--- a/example/tknewsbiff
+++ b/example/tknewsbiff
@@ -4,7 +4,10 @@
 exec tclsh "$0" ${1+"$@"}
 
 package require Expect
-package require Tk
+if {[catch {package require Tk}]} {
+    puts stderr "This script require Tk to run. You can install the tk package for that."
+    exit 1
+}
 
 # Name: tknewsbiff
 # Author: Don Libes
--- a/example/xpstat
+++ b/example/xpstat
@@ -4,6 +4,10 @@
 exec tclsh "$0" ${1+"$@"}
 
 package require Expect
+if {[catch {package require Tk}]} {
+    puts stderr "This script require Tk to run. You can install the tk package for that."
+    exit 1
+}
 
 
 # This script acts as a front-end for xpilot.  Run it in the background,
