diff --git a/tclconfig/tcl.m4 b/tclconfig/tcl.m4
index 0689cab..5a4b54c 100644
--- a/tclconfig/tcl.m4
+++ b/tclconfig/tcl.m4
@@ -2509,7 +2509,6 @@ AC_DEFUN([TEA_TCL_LINK_LIBS], [
     #--------------------------------------------------------------------
 
     AC_CHECK_FUNC(sin, MATH_LIBS="", MATH_LIBS="-lm")
-    AC_CHECK_LIB(ieee, main, [MATH_LIBS="-lieee $MATH_LIBS"])
 
     #--------------------------------------------------------------------
     # Interactive UNIX requires -linet instead of -lsocket, plus it
